### ISSUE TITLE
MSVS_2017 and 2019 failures on 2 windows machines

### ISSUE THREAD

**Issue by Haroon-Khel** [adoptium/infrastructure#2317]
ref https://github.com/adoptium/temurin-build/pull/2693#issuecomment-918463457

Errors when trying to install the 2017 and 2019 compilers on to test-azure-win2012r2-x64-1  and test-packet-win2012r2-x64-1
 

test-azure-win2012r2-x64-1 

```
TASK [MSVS_2017 : Register Visual Studio Community 2017 DIA SDK shared libraries] ***
failed: [test-azure-win2012r2-x64-1] (item=C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\DIA SDK\bin\msdia140.dll) => {"ansible_loop_var": "item", "changed": true, "cmd": "regsvr32 /s \"C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\Community\\DIA SDK\\bin\\msdia140.dll\"", "delta": "0:00:00.056998", "end": "2021-09-14 11:51:46.972891", "item": "C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\Community\\DIA SDK\\bin\\msdia140.dll", "msg": "non-zero return code", "rc": 3, "start": "2021-09-14 11:51:46.915892", "stderr": "", "stderr_lines": [], "stdout": "", "stdout_lines": []}
failed: [test-azure-win2012r2-x64-1] (item=C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\DIA SDK\bin\amd64\msdia140.dll) => {"ansible_loop_var": "item", "changed": true, "cmd": "regsvr32 /s \"C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\Community\\DIA SDK\\bin\\amd64\\msdia140.dll\"", "delta": "0:00:00.047000", "end": "2021-09-14 11:51:48.316016", "item": "C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\Community\\DIA SDK\\bin\\amd64\\msdia140.dll", "msg": "non-zero return code", "rc": 3, "start": "2021-09-14 11:51:48.269016", "stderr": "", "stderr_lines": [], "stdout": "", "stdout_lines": []}
```

test-packet-win2012r2-x64-1

```
TASK [MSVS_2019 : Register Visual Studio Community 2019 DIA SDK shared libraries] ***
failed: [test-packet-win2012r2-x64-1] (item=C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\DIA SDK\bin\msdia140.dll) => {"ansible_loop_var": "item", "changed": true, "cmd": "regsvr32 /s \"C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Community\\DIA SDK\\bin\\msdia140.dll\"", "delta": "0:00:00.159196", "end": "2021-09-13 06:14:09.241804", "item": "C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Community\\DIA SDK\\bin\\msdia140.dll", "msg": "non-zero return code", "rc": 3, "start": "2021-09-13 06:14:09.082608", "stderr": "", "stderr_lines": [], "stdout": "", "stdout_lines": []}
failed: [test-packet-win2012r2-x64-1] (item=C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\DIA SDK\bin\amd64\msdia140.dll) => {"ansible_loop_var": "item", "changed": true, "cmd": "regsvr32 /s \"C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Community\\DIA SDK\\bin\\amd64\\msdia140.dll\"", "delta": "0:00:00.148142", "end": "2021-09-13 06:14:12.750131", "item": "C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Community\\DIA SDK\\bin\\amd64\\msdia140.dll", "msg": "non-zero return code", "rc": 3, "start": "2021-09-13 06:14:12.601989", "stderr": "", "stderr_lines": [], "stdout": "", "stdout_lines": []}
```

**Comment by Haroon-Khel** [comment]
test-packet-win2012r2-x64-1 looks good. Had to install some extra components using the MSVS 2019 community edition installer, found in the control panel. OpenSSL 2019 was installed fine onto the machine.

The same solution sort of worked with test-azure-win2012r2-x64-1, using the MSVS 2017 community edition installer, but OpenSSL 2017 still hit an error when installing

**Comment by sxa** [comment]
> OpenSSL 2017 still hit an error when installing

Installing as opposed to the compilation phase? What was the error?

**Comment by Haroon-Khel** [comment]
```
{
    "start": "2021-09-28 11:17:41.910735",
    "stdout": "**********************************************************************\r\n** Visual Studio 2017 Developer Command Prompt v15.9.39\r\n** Copyright (c) 2017 Microsoft Corporation\r\n**********************************************************************\r\n[vcvarsall.bat] Environment initialized for: 'x64'\r\nConfiguring OpenSSL version 1.1.1l (0x101010cfL) for VC-WIN64A\r\nUsing os-specific seed configuration\r\nCreating configdata.pm\r\nCreating makefile\r\n\r\n**********************************************************************\r\n***                                                                ***\r\n***   OpenSSL has been successfully configured                     ***\r\n***                                                                ***\r\n***   If you encounter a problem while building, please open an    ***\r\n***   issue on GitHub <https://github.com/openssl/openssl/issues>  ***\r\n***   and include the output from the following command:           ***\r\n***                                                                ***\r\n***       perl configdata.pm --dump                                ***\r\n***                                                                ***\r\n***   (If you are new to OpenSSL, you might want to consult the    ***\r\n***   'Troubleshooting' section in the INSTALL file first)         ***\r\n***                                                                ***\r\n**********************************************************************\r\n",
    "cmd": "set PATH=C:\\Strawberry\\perl\\bin;C:\\openjdk\\nasm-2.13.03;%PATH% && .\\vcvarsall.bat AMD64 && cd C:\\temp\\OpenSSL-1.1.1l && perl C:\\temp\\OpenSSL-1.1.1l\\Configure VC-WIN64A --prefix=C:\\openjdk\\OpenSSL-1.1.1l-x86_64-VS2017 && nmake install > C:\\temp\\openssl64-VS2017.log && nmake -f makefile clean",
    "msg": "non-zero return code",
    "stderr": "\r\nMicrosoft (R) Program Maintenance Utility Version 14.16.27045.0\r\nCopyright (C) Microsoft Corporation.  All rights reserved.\r\n\r\n\r\nMicrosoft (R) Program Maintenance Utility Version 14.16.27045.0\r\nCopyright (C) Microsoft Corporation.  All rights reserved.\r\n\r\n\r\nMicrosoft (R) Program Maintenance Utility Version 14.16.27045.0\r\nCopyright (C) Microsoft Corporation.  All rights reserved.\r\n\r\nNMAKE : fatal error U1077: '\"C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\Community\\VC\\Tools\\MSVC\\14.16.27023\\bin\\HostX64\\x64\\cl.EXE\"' : return code '0x2'\r\nStop.\r\nNMAKE : fatal error U1077: '\"C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\Community\\VC\\Tools\\MSVC\\14.16.27023\\bin\\HostX64\\x64\\nmake.exe\"' : return code '0x2'\r\nStop.\r\n",
    "changed": true,
    "rc": 2,
    "delta": "0:00:04.361455",
    "end": "2021-09-28 11:17:46.272191",
    "stdout_lines": [
        "**********************************************************************",
        "** Visual Studio 2017 Developer Command Prompt v15.9.39",
        "** Copyright (c) 2017 Microsoft Corporation",
        "**********************************************************************",
        "[vcvarsall.bat] Environment initialized for: 'x64'",
        "Configuring OpenSSL version 1.1.1l (0x101010cfL) for VC-WIN64A",
        "Using os-specific seed configuration",
        "Creating configdata.pm",
        "Creating makefile",
        "",
        "**********************************************************************",
        "***                                                                ***",
        "***   OpenSSL has been successfully configured                     ***",
        "***                                                                ***",
        "***   If you encounter a problem while building, please open an    ***",
        "***   issue on GitHub <https://github.com/openssl/openssl/issues>  ***",
        "***   and include the output from the following command:           ***",
        "***                                                                ***",
        "***       perl configdata.pm --dump                                ***",
        "***                                                                ***",
        "***   (If you are new to OpenSSL, you might want to consult the    ***",
        "***   'Troubleshooting' section in the INSTALL file first)         ***",
        "***                                                                ***",
        "**********************************************************************"
    ],
    "stderr_lines": [
        "",
        "Microsoft (R) Program Maintenance Utility Version 14.16.27045.0",
        "Copyright (C) Microsoft Corporation.  All rights reserved.",
        "",
        "",
        "Microsoft (R) Program Maintenance Utility Version 14.16.27045.0",
        "Copyright (C) Microsoft Corporation.  All rights reserved.",
        "",
        "",
        "Microsoft (R) Program Maintenance Utility Version 14.16.27045.0",
        "Copyright (C) Microsoft Corporation.  All rights reserved.",
        "",
        "NMAKE : fatal error U1077: '\"C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\Community\\VC\\Tools\\MSVC\\14.16.27023\\bin\\HostX64\\x64\\cl.EXE\"' : return code '0x2'",
        "Stop.",
        "NMAKE : fatal error U1077: '\"C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\Community\\VC\\Tools\\MSVC\\14.16.27023\\bin\\HostX64\\x64\\nmake.exe\"' : return code '0x2'",
        "Stop."
    ],
    "_ansible_no_log": false
```

**Comment by sxa** [comment]
Hmmm not much info there- I'd  suggest going onto the machine and runing the commands in the openssl step manually and seeing if you can get anything more out of it, particularly if you can figure out the full `cl` command used.

**Comment by Haroon-Khel** [comment]
I think I've fixed it. The 2017 compile was missing some tools, which I installed through the community installer. This seems to be a common error with the installation of our windows compilers, a absence of some of the necessary tools which need to be installed manually. I'll look for a way to automate this.

In any case, OpenSSL 2017 installed without error. Ive hit a hurdle with the 2019 OpenSSL installation which I hope can be resolved in the same way
